src/java/org/apache/cassandra/index/SecondaryIndexManager.java

```diff
@@ -619,8 +642,16 @@ private synchronized void markIndexesBuilding(Set<Index> indexes, boolean isFull
     private synchronized void markIndexBuilt(Index index, boolean isFullRebuild)
     {
         String indexName = index.getIndexMetadata().name;
-        if (isFullRebuild)
+        if (isFullRebuild && index.supportsLoad(Index.LoadType.READ))
+        {
             queryableIndexes.add(indexName);
+            logger.info("Index [" + indexName + "] became queryable.");
+        }
+        if (isFullRebuild && index.supportsLoad(Index.LoadType.WRITE))
+        {
+            writableIndexes.add(indexName);
+            logger.info("Index [" + indexName + "] became writable.");
```

Comment:
I think that `became` suggests a change of status, as if it weren't writable before. We could either say that is writable without telling if it was so before or, even better, use the boolean returned by `queryableIndexes.add`:
```suggestion
            if (writableIndexes.add(indexName))
                logger.info("Index [" + indexName + "] became writable.");
```
The same applies to queryable indexes above.